service/src/main/java/bio/terra/pipelines/service/ImputationService.java

```diff
@@ -131,15 +130,50 @@ public List<ListAppResponse> queryForWorkspaceApps() {
           imputationConfiguration.workspaceId(),
           getAppsResponse);
 
+      //      // example to create cromwell_runner_app
+      //      CreateAppRequest createAppRequest =
+      //          new CreateAppRequest()
+      //              .appType(AppType.CROMWELL_RUNNER_APP)
+      //              .accessScope(AppAccessScope.USER_PRIVATE)
+      //              .labels(Map.of("saturnAutoCreated", "true"));
+      //      logger.info(
+      //          "creating runner app for workspace {}: {}",
+      //          workspaceId,
+      //          leonardoService.createAppV2(
+      //              workspaceId,
+      //              samService.getTspsServiceAccountToken(),
+      //              "tsps-cr-" + UUID.randomUUID(),
+      //              createAppRequest));
+
       // wds related calls
       String wdsUri =
           leonardoService.getWdsUrlFromApps(
               workspaceId, samService.getTspsServiceAccountToken(), false);
+
       logger.info("Wds uri for workspace id {}: {}", imputationConfiguration.workspaceId(), wdsUri);
+
       logger.info(
           "Wds health: {}",
           wdsService.checkHealth(wdsUri, samService.getTspsServiceAccountToken()));
 
+      String id = UUID.randomUUID().toString();
```

Comment:
are we generating a random UUID here because we may end up retrying this step? even so, I'd think it would be ok to use the same UUID so we know we end up with a single row in the table. why not use the flightId here?

Reply:
yea when we're doing it in a step we will use the flight id here, cant do that here because this isnt a step. I also want to pull stuff out of this function, not put more stuff in.

Reply:
fair enough. maybe just leave a comment here to change this to flight id when it goes into a step so we don't forget?